sdk/python/kfp/v2/components/components.py

```diff
@@ -19,39 +19,39 @@
     'load_component_from_file',
 ]
 
-import copy
-import sys
-from collections import OrderedDict
-from typing import Any, Callable, List, Mapping, NamedTuple, Sequence, Union
+import collections
 import inspect
-from kfp.components import _dynamic
+from typing import Any, Callable, List, Mapping, NamedTuple, Sequence, Union
 
-from kfp.components._naming import _sanitize_file_name, _sanitize_python_function_name, generate_unique_name_conversion_table
-from kfp.components._yaml_utils import load_yaml
-from kfp.components._data_passing import serialize_value, get_canonical_type_for_type_struct
+from kfp.components import _data_passing
+from kfp.components import _dynamic
+from kfp.components import _naming
+from kfp.components import _yaml_utils
 from kfp.v2.components import structures
 
 _default_component_name = 'Component'
 
 
 def load_component(filename=None, url=None, text=None):
-  """Loads component from text, file or URL and creates a task factory function
+  """Loads component from text, file or URL and creates a task factory function.
 
-    Only one argument should be specified.
+  Only one argument should be specified.
 
-    Args:
-        filename: Path of local file containing the component definition.
-        url: The URL of the component file data.
-        text: A string containing the component file data.
+  Args:
+    filename: Path of local file containing the component definition.
+    url: The URL of the component file data.
+    text: A string containing the component file data.
 
-    Returns:
-        A factory function with a strongly-typed signature.
-        Once called with the required arguments, the factory constructs a
-        pipeline task instance (ContainerOp).
-    """
-  #This function should be called load_task_factory since it returns a factory function.
-  #The real load_component function should produce an object with component properties (e.g. name, description, inputs/outputs).
-  #TODO: Change this function to return component spec object but it should be callable to construct tasks.
+  Returns:
+    A factory function with a strongly-typed signature.
+    Once called with the required arguments, the factory constructs a
+    pipeline task instance (ContainerOp).
+  """
+  # This function should be called load_task_factory since it returns a factory
```

Comment:
If indeed it worth considering rename this function, then perhaps we shall leave a TODO here.

Up to you

Reply:
I'll probably remove this comment. It's from 2 years ago. And `load_component*` are widely used public functions, so I don't think we would ever make this breaking change. 
PS: I think the naming here is fine. "load_something" is more about loading something that is an input rather than the return.

Reply:
removed the comment